### PR TITLE
@l2succes: adds analytics for mobile auth cta

### PR DESCRIPTION
**Pull request by xtina-starr**
Addresses https://artsyproduct.atlassian.net/browse/GROW-880.

Adds analytics events:
- Auth impression when the cta banner is displayed
- Click event for when the user dismiss the cta banner
- Passes analytic properties to auth sign up modal

### CODE REVIEW

**Comment by eessex** [comment]
@alloy yeah good point -- we should be adding new events especially to the analytics schema, and converting old ones as we touch them.  Happy to pair on this if you have q's.

File: src/Components/MinimalCtaBanner.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { Sans } from "@artsy/palette"
 import React from "react"
+import track from "react-tracking"
```
Comment (l2succes) [inline src/Components/MinimalCtaBanner.tsx]: We should use our typed version of `track` here
https://github.com/artsy/reaction/blob/master/src/Artsy/Analytics/track.ts

and add approriate values to the schema (https://github.com/artsy/reaction/blob/master/src/Artsy/Analytics/Schema/Values.ts#L70) if needed

File: src/Components/MinimalCtaBanner.tsx [hunk 2]
```diff
@@ -26,7 +30,27 @@ export class MinimalCtaBanner extends React.Component<
     dismissed: false,
   }
 
+  componentDidMount() {
+    if (this.props.tracking) {
+      this.props.tracking.trackEvent({
```
Comment (l2succes) [inline src/Components/MinimalCtaBanner.tsx]: use `@track` decorator

**Comment by xtina-starr** [comment]
I made some changes based on the feedback provided but I did have questions regarding the difference in schema that I was provided in the ticket and the way we are approaching this in reaction. I left a comment in the [ticket](https://artsyproduct.atlassian.net/browse/GROW-880) itself so feel free to chime in.

File: src/Artsy/Analytics/Schema/Interaction.ts [hunk 3]
```diff
@@ -35,3 +35,22 @@ export interface Interaction {
...
+  */
+  flow?: string
+
+  /*
+  *  The type of action that triggered the modal (eg: click, timed)
+  */
+  trigger?: string
+}
+
+export type Interaction = BaseInteraction & AuthInteraction
```
Comment (alloy) [inline src/Artsy/Analytics/Schema/Interaction.ts]: I don’t think this is right, as it effectively adds these 3 optional properties to all places where an `Interaction` is used. I think you should export both, have `AuthInteraction` extend `BaseInteraction`, and in `src/Artsy/Analytics/Schema/index.ts` add `AuthInteraction` to the `Trackables` union.

I’m also not a big fan of naming things “base”, I would prefer `BaseInteraction` were named after what it’s used for. (The same applies to `AuthInteraction` btw, which I _think_ may be an interaction that led to an authentication prompt?)

I also wonder if indeed `intent` should really be optional or if you only had to do that to make type-checking work, which would presumably fail on all places where `Interaction` was already being used without `intent`? Once you make `AuthInteraction` its own thing (that extends `BaseInteraction`), this problem would go away and allow you to have required properties.

Comment (xtina-starr) [reply src/Artsy/Analytics/Schema/Interaction.ts]: @l2succes and I actually discussed this approach. I can't remember why we decided to combined the types though. I do think it makes sense to export `AuthenticationInteraction` separately and add it to the `Trackables` union. Luc do you have any objection?

`intent` is used just about everywhere except for when the CTA Banner is dismissed. For that reason I think we would want to make it optional.

**Comment by xtina-starr** [comment]
@l2succes @l2succes wondering if there is any additional feedback for this PR?

**Comment by l2succes** [comment]
@xtina-starr Just left a comment about the signup URL string building. Everything else looks good

**Pull request merged**